pyproject.toml

```diff
@@ -32,6 +32,7 @@ python-dateutil = ">=2.8.2"
 urllib3 = ">=1.26.8"
 certifi = ">=2021.10.8"
 pytz = ">=2021.3"
+PyYAML = ">=6.0"
```

Comment:
Should we make this an optional dependency? If pyyaml is available we can enable the to_yaml method otherwise we can throw an exception asking users to install `hera-workflows[yaml]`. Additionally maybe we can add a to_json method?

Reply:
See comment on optionality [here](https://github.com/argoproj-labs/hera-workflows/pull/327#discussion_r1007532840) and [this one](https://github.com/argoproj-labs/hera-workflows/pull/327#discussion_r1007529489) for `to_json`!